search-app/src/cmr/search/services/query_execution/has_granules_or_cwic_results_feature.clj

```diff
@@ -0,0 +1,87 @@
+  "The frequency in seconds of the refresh-has-granules-or-cwic-map-job"
+  ;; default to 1 hour
+  3600)
+
+(def has-granule-cache-key
+  :has-granules-or-cwic-map)
+
+(defconfig cwic-tag
+  "has-granules-or-cwic should also return any collection with configured cwic-tag"
+  {:default "CWIC"})
+
+(defn create-has-granules-or-cwic-map-cache
+  "Returns a 'cache' which will contain the cached has granules map."
+  []
+  (mem-cache/create-in-memory-cache))
+
+(defn get-cwic-collections
+  "Returns the collection granule count by searching elasticsearch by aggregation"
+  [context provider-ids]
+  (let [condition (nf/parse-nested-condition :tags {:tag-key (cwic-tag)} false true)
+        query (qm/query {:concept-type :collection
+                         :condition condition})
+        results (common-esi/execute-query context query)]
+    (into {}
+          (for [coll-id (map :_id (get-in results [:hits :hits]))]
+            [coll-id 1]))))
+
+(defn- collection-granule-counts->has-granules-or-cwic-map
+  "Converts a map of collection ids to granule counts to a map of collection ids to true or false
+  of whether the collection has any granules"
+  [coll-gran-counts]
+  (into {} (for [[coll-id num-granules] coll-gran-counts]
+             [coll-id (> num-granules 0)])))
+
+(defn refresh-has-granules-or-cwic-map
+  "Gets the latest provider holdings and updates the has-granules-or-cwic-map stored in the cache."
+  [context]
+  (let [has-granules-or-cwic-map (collection-granule-counts->has-granules-or-cwic-map
+                                  (merge
+                                   (idx/get-collection-granule-counts context nil)
+                                   (get-cwic-collections context nil)))]
+    (cache/set-value (cache/context->cache context has-granule-cache-key)
+                     :has-granules-or-cwic has-granules-or-cwic-map)))
+
+(defn get-has-granules-or-cwic-map
+  "Gets the cached has granules map from the context which contains collection ids to true or false
+  of whether the collections have granules or not. If the has-granules-or-cwic-map has not yet been cached
+  it will retrieve it and cache it."
+  [context]
+  (let [has-granules-or-cwic-map-cache (cache/context->cache context has-granule-cache-key)]
+    (cache/get-value has-granules-or-cwic-map-cache
+                     :has-granules-or-cwic
+                     (fn []
+                       (collection-granule-counts->has-granules-or-cwic-map
+                        (merge
+                         (idx/get-collection-granule-counts context nil)
+                         (get-cwic-collections context nil)))))))
+
+;; This returns a boolean flag with collection results if a collection has any granules in provider holdings
+(defmethod query-execution/post-process-query-result-feature :has-granules-or-cwic
+  [context query elastic-results query-results feature]
+  (assoc query-results :has-granules-or-cwic-map (get-has-granules-or-cwic-map context)))
```

Comment:
Are we returning this field and value in the JSON response? I'm thinking we just need to be able to search by it, not return a field in the response format.

Reply:
No, I was starting to change all the result handlers then I realized that I probably didn't need to.  Forgot to remove this.